src/hint.c

```diff
@@ -39,16 +39,26 @@ size_t generate_hints(screen_t scr, struct hint *hints)
 
 	screen_get_dimensions(scr, &sw, &sh);
 
-	const int colgap = 300/cfg->hint_size;
-	const int rowgap = 300/cfg->hint_size;
+        /* hint_size corresponds to the percentage of column/row space
+         * taken up by the hints. At the cost of including math.h, one
+         * could replace cfg->hint-size / 100 by its square root, so
+         * that hint_size corresponds to the approximate percentage of
+         * screen area taken up by the hints
+         */
 
-	const int w = sw / nc - colgap;
-	const int h = sh / nr - rowgap;
+        const int w = sw / nc * cfg->hint_size / 100;
+        const int h = sh / nr * cfg->hint_size / 100;
```

Comment:
This would again result in hint_size being a value with no simple interpretation (currently it is a percentage of covered horizontal/vertical space). Still I'm happy to do it if you think that minimizing surprises is more important.

Reply:
One could argue it is a scaled percentage, but I'm not much fussed about the units since most people will probably just toy with the value until they find something they are happy with. Having said that, it's probably reasonable to change it this once.

Reply:
Sorry but I don't understand your last sentence (English is not my mother tongue), would you like to keep it at 100 or change it to ~40? I'm always happy when I can attach some meaning to a value, although you're perfectly right when saying that people would nevertheless toy around until finding a sweet spot, but maybe with some better initial value if the percentage interpretation is given...

Reply:
Sorry, there was some ambiguity in my response. Leaving it as is (a percentage) should be fine. You can ignore my earlier suggestion